MeshLib/Node.h

```diff
@@ -79,13 +70,16 @@ class Node : public MathLib::Point3dWithID
 		return Node(_x[0]-v[0], _x[1]-v[1], _x[2]-v[2]);
 	}
 
-protected:
-	/**
-	 * Add an element the node is part of.
-	 * This method is called by Mesh::addElement(Element*), see friend definition.
-	 */
+	/// Update coordinates of a node.
+	/// This method automatically also updates the areas/volumes of all connected elements.
+	virtual void updateCoordinates(double x, double y, double z);
+
+	/// Add an element the node is part of.
 	void addElement(Element* elem) { _elements.push_back(elem); }
 
+	/// clear stored elements connecting to this node
+	void clearElements() { _elements.clear(); }
```

Comment:
Since in my opinion the method should be used with special caution, I wonder if it is necessary to make this method public? This opens the possibility to delete to the elements from all places within OGS.